src/main/features/ccj/form/models/repaymentPlan.ts

```diff
@@ -0,0 +1,76 @@
+import { LocalDate } from 'forms/models/localDate'
+import { PaymentSchedule } from 'ccj/form/models/paymentSchedule'
+import { IsDefined, IsIn, IsPositive, ValidateNested } from 'class-validator'
+import { isValidYearFormat } from 'forms/validation/validators/isValidYearFormat'
+import { IsValidLocalDate } from 'forms/validation/validators/isValidLocalDate'
+import { IsFutureDate } from 'app/forms/validation/validators/dateFutureConstraint'
+import { IsLessThan } from 'forms/validation/validators/isLessThan'
+
+export class ValidationErrors {
+  static readonly FIRST_PAYMENT_AMOUNT_INVALID: string = 'Enter a valid amount of first payment'
+  static readonly INSTALMENTS_AMOUNT_INVALID: string = 'Enter a valid amount for equal instalments'
+  static readonly FUTURE_DATE: string = 'Enter a first payment date in the future'
+  static readonly INVALID_DATE: string = 'Enter a valid date of first payment'
+  static readonly SELECT_PAYMENT_SCHEDULE: string = 'Select how often they should pay'
+}
+
+export class RepaymentPlan {
+
+  remainingAmount?: number
+
+  @IsPositive({ message: ValidationErrors.FIRST_PAYMENT_AMOUNT_INVALID })
+  @IsLessThan('remainingAmount', { message: ValidationErrors.FIRST_PAYMENT_AMOUNT_INVALID })
+  firstPayment?: number
+
+  @IsPositive({ message: ValidationErrors.INSTALMENTS_AMOUNT_INVALID })
+  @IsLessThan('remainingAmount', { message: ValidationErrors.INSTALMENTS_AMOUNT_INVALID })
+  installmentAmount?: number
+
+  @ValidateNested()
+  @IsDefined({ message: ValidationErrors.INVALID_DATE })
+  @IsValidLocalDate({ message: ValidationErrors.INVALID_DATE })
+  @isValidYearFormat(4, { message: ValidationErrors.INVALID_DATE })
```

Comment:
Please see `src/main/app/forms/models/address.ts/ValidationConstants` as an example of constants for validation.

It's easier to make unit tests maintainable.

Reply:
I removed the unneeded configuration value instead